File: windowsversion/src/types/timeline.ts
/** 时间轴 — 有录音时自动生成 */
export interface Timeline {
  id: string;
  projectId: string;
  startTime: string;
  endTime: string;
  duration: number;
  audioTracks: AudioTrack[];
  keyframes: Keyframe[];
  markers: Marker[];
}

/** 音频轨道 */
export interface AudioTrack {
  id: string;
  timelineId: string;
  filePath: string;
  fileName: string;
  format: AudioFormat;
  duration: number;
  sampleRate: number;
  channels: number;
  waveformData: string;
  transcription: Transcription | null;
  offsetInTimeline: number;
}

export type AudioFormat = "wav" | "m4a" | "mp3" | "aac";

/** AI 转录结果 */
export interface Transcription {
  id: string;
  audioTrackId: string;
  language: string;
  segments: TranscriptionSegment[];
  status: TranscriptionStatus;
}

export type TranscriptionStatus =
  | "pending"
  | "processing"
  | "completed"
  | "failed";

/** 转录片段 */
export interface TranscriptionSegment {
  startTime: number;
  endTime: number;
  text: string;
  confidence: number;
  speaker: string | null;
}

/** 关键帧 — 素材在时间轴上的锚定 */
export interface Keyframe {
  id: string;
  timelineId: string;
  assetId: string;
  anchorTime: number;
  liveAudioClipId: string | null;
  source: "auto" | "manual";
}

/** 时间标记 */
export interface Marker {
  id: string;
  timelineId: string;
  time: number;
  label: string;
  color: string;
  type: MarkerType;
}

export type MarkerType = "bookmark" | "chapter" | "important" | "question";

File: windowsversion/src-tauri/src/models/timeline.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Timeline {
    pub id: String,
    pub project_id: String,
    pub start_time: String,
    pub end_time: String,
    pub duration: f64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct AudioTrack {
    pub id: String,
    pub timeline_id: String,
    pub file_path: String,
    pub file_name: String,
    pub format: String,
    pub duration: f64,
    pub sample_rate: i64,
    pub channels: i64,
    pub waveform_data: String,
    pub offset_in_timeline: f64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Transcription {
    pub id: String,
    pub audio_track_id: String,
    pub language: String,
    pub segments_json: String,
    pub status: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Keyframe {
    pub id: String,
    pub timeline_id: String,
    pub asset_id: String,
    pub anchor_time: f64,
    pub live_audio_clip_id: Option<String>,
    pub source: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Marker {
    pub id: String,
    pub timeline_id: String,
    pub time: f64,
    pub label: String,
    pub color: String,
    pub marker_type: String,
}
